rust/extensions.bzl

```
@@ -80,6 +80,14 @@ def _rust_impl(module_ctx):
         fail("Multiple host_tools were defined in your root MODULE.bazel")
 
     host_triple = get_host_triple(module_ctx)
+
+    iso_date = None
+    version = host_tools.version or rust_common.default_version
+    version_array = version.split("/")
+    if len(version_array) > 1:
+        version = version_array[0]
+        iso_date = version_array[1]
```

Perhaps slightly cleaner
```suggestion
    # Any version containing a slash is expected to be a nightly/beta release with iso date. E.g. `nightly/2024-03-21`
    if "/" in version:
        version, _, iso_date = version.partition("/")
```

---

This is probably gonna bite me but gonna try to yolo-merge 😅